### PR TITLE
Make nightly clippy happy

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

N/A

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/file_format/mod.rs [hunk 1]
```diff
@@ -520,6 +520,7 @@ impl From<ObjectMeta> for FileMeta {
 ///
 ///              ParquetExec
 ///```
+#[allow(clippy::manual_filter)]
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/src/physical_plan/file_format/mod.rs:523]: 👍  hopefully they'll fix that prior to release

**Review by alamb — APPROVED** [review]

File: datafusion/core/src/physical_plan/file_format/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (xudong963) [reply datafusion/core/src/physical_plan/file_format/mod.rs:523]: It seems the latest nightly fixes it. 

However, the latest nightly introduced: https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args. Our codebase needs to change many many places 😭 

I'll continue to update the ticket when I don't want to use my brain 🤣 !

**Comment by xudong963** [comment]
TIL: `cargo clippy --fix`.  I like it 🤣

**Comment by alamb** [comment]
>  TIL: cargo clippy --fix. I like it 🤣

😍

**Pull request merged**